core/src/main/java/ai/timefold/solver/core/impl/localsearch/decider/acceptor/lateacceptance/DiversifiedLateAcceptanceAcceptor.java

```diff
@@ -0,0 +1,113 @@
+package ai.timefold.solver.core.impl.localsearch.decider.acceptor.lateacceptance;
+
+import java.util.Arrays;
+
+import ai.timefold.solver.core.api.score.Score;
+import ai.timefold.solver.core.impl.localsearch.decider.acceptor.AbstractAcceptor;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchMoveScope;
+import ai.timefold.solver.core.impl.localsearch.scope.LocalSearchPhaseScope;
+
+public class DiversifiedLateAcceptanceAcceptor<Solution_> extends AbstractAcceptor<Solution_> {
+
+    // The worst score in the late elements list
+    protected Score<?> lateWorse;
+    // Number of occurrences of lateWorse in the late elements
+    protected int lateWorseOccurrences = -1;
+
+    protected int lateAcceptanceSize = -1;
+
+    protected Score<?>[] previousScores;
+    protected int lateScoreIndex = -1;
+
+    public void setLateAcceptanceSize(int lateAcceptanceSize) {
+        this.lateAcceptanceSize = lateAcceptanceSize;
+    }
+
+    // ************************************************************************
+    // Worker methods
+    // ************************************************************************
+
+    @Override
+    public void phaseStarted(LocalSearchPhaseScope<Solution_> phaseScope) {
+        super.phaseStarted(phaseScope);
+        validate();
+        previousScores = new Score[lateAcceptanceSize];
+        var initialScore = phaseScope.getBestScore();
+        Arrays.fill(previousScores, initialScore);
+        lateScoreIndex = 0;
+        lateWorseOccurrences = lateAcceptanceSize;
+        lateWorse = initialScore;
+    }
+
+    private void validate() {
+        if (lateAcceptanceSize <= 0) {
+            throw new IllegalArgumentException(
+                    "The lateAcceptanceSize (%d) cannot be negative or zero.".formatted(lateAcceptanceSize));
+        }
+    }
+
+    @Override
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    public boolean isAccepted(LocalSearchMoveScope<Solution_> moveScope) {
+        // The acceptance and replacement strategies are based on the work:
+        // Diversified Late Acceptance Search by M. Namazi, C. Sanderson, M. A. H. Newton, M. M. A. Polash, and A. Sattar
+        var lateScore = previousScores[lateScoreIndex];
```

Comment:
Please declare variables closer to where you need them, you decrease the size of the context readers need to keep in their heads.
This particular variable is only needed on line 63.